### PR TITLE
Resolve pyright errors in Loaders

### LINKED ISSUE #585: Resolve Pyright errors for Loaders

### PR DESCRIPTION
**Pull request by andrewfrench**
Resolves #585

### CODE REVIEW

File: griptape/loaders/base_text_loader.py
Comment (andrewfrench) [inline griptape/loaders/base_text_loader.py:1]: Introduced `BaseTextLoader` to address method override issues surrounding `load()` on loaders that previous inherited from `TextLoader`.

File: griptape/loaders/email_loader.py [hunk 1]
```diff
@@ -57,21 +60,28 @@ def _retrieve_email(self, query: EmailQuery) -> ListArtifact | ErrorArtifact:
                 if key and search_criteria:
                     _typ, [message_numbers] = client.search(None, key, f'"{search_criteria}"')
                     messages_count = self._count_messages(message_numbers)
-                else:
+                elif len(mailbox) > 1 and len(mailbox[1]) > 0 and mailbox[1][0] is not None:
```
Comment (collindutter) [inline griptape/loaders/email_loader.py:63]: For `if len(my_list) > 0` checks we can just do `if my_list`.

File: griptape/loaders/email_loader.py [hunk 2]
```diff
@@ -57,21 +60,28 @@ def _retrieve_email(self, query: EmailQuery) -> ListArtifact | ErrorArtifact:
...
                     messages_count = int(mailbox[1][0])
+                else:
+                    raise Exception("unable to parse number of messages")
 
                 top_n = max(0, messages_count - max_count) if max_count else 0
                 for i in range(messages_count, top_n, -1):
                     result, data = client.fetch(str(i), "(RFC822)")
+
+                    if data is None or len(data) == 0 or data[0] is None:
```
Comment (collindutter) [inline griptape/loaders/email_loader.py:72]: `len(data)` -> `not data`

**Review by collindutter — CHANGES_REQUESTED** [review]
Couple minor things, otherwise looks great.

**Review by collindutter — APPROVED** [review]

**Pull request merged**